modules/mono/editor/GodotTools/GodotTools/Build/BuildManager.cs

```diff
@@ -192,6 +196,87 @@ private static bool BuildProjectBlocking(BuildInfo buildInfo)
             return true;
         }
 
+        public static bool Publish(BuildInfo buildInfo)
+        {
+            if (_buildInProgress != null)
+                throw new InvalidOperationException("A build is already in progress");
+
+            _buildInProgress = buildInfo;
+
+            try
+            {
+                BuildStarted?.Invoke(buildInfo);
+
+                // Required in order to update the build tasks list
+                Internal.GodotMainIteration();
+
+                try
+                {
+                    RemoveOldIssuesFile(buildInfo);
+                }
+                catch (IOException e)
+                {
+                    BuildLaunchFailed?.Invoke(buildInfo, $"Cannot remove issues file: {GetIssuesFilePath(buildInfo)}");
+                    Console.Error.WriteLine(e);
+                }
+
+                try
+                {
+                    int exitCode = BuildSystem.Publish(buildInfo, StdOutputReceived, StdErrorReceived);
+
+                    if (exitCode != 0)
+                        PrintVerbose(
+                            $"dotnet publish exited with code: {exitCode}. Log file: {GetLogFilePath(buildInfo)}");
+
+                    BuildFinished?.Invoke(exitCode == 0 ? BuildResult.Success : BuildResult.Error);
+
+                    return exitCode == 0;
+                }
+                catch (Exception e)
+                {
+                    BuildLaunchFailed?.Invoke(buildInfo,
+                        $"The publish method threw an exception.\n{e.GetType().FullName}: {e.Message}");
+                    Console.Error.WriteLine(e);
+                    return false;
+                }
+            }
+            finally
+            {
+                _buildInProgress = null;
+            }
+        }
+
+        public static bool PublishBlocking(
+            [NotNull] string config,
+            [NotNull] string platform,
+            [NotNull] string rid,
+            string publishOutputDir
+        )
+        {
+            var buildInfo = new BuildInfo(GodotSharpDirs.ProjectSlnPath,
+                targets: new string[] { } /* Ignored when publishing */,
```

Comment:
We may want to promote [`CA1825`](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1825) to **warning** (defaults to **suggestion**) so it shows up in CI.

Reply:
We should probably add `<AnalysisMode>Recommended</AnalysisMode>` to more projects as we already did to `GodotSharp.csproj`. But that's out of the scope of this PR, as it will likely require some work fixing code or manually disabling possible new unwanted warnings.
Regarding the requested change, I removed the parameter after all.